src/Questionnaire/UpdateQuestionnaire.js

```diff
@@ -0,0 +1,138 @@
+import { getCookie } from "../cookies";
+import Header from "../Header/Header";
+
+class UpdateQuestionnaire extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: getCookie("userId"),
+      questionBank: [],
+      response: [],
+      error:"",
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    if (this.checkResponses()) {
+      //checks if all questions have been answered
+        const id = this.state.id;
+        const myRequest = new Request("http://127.0.0.1:5000/updateQuestionnaire", {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify({ responses: this.state.response, userId: id }),
+        });
+          const matchRequest = new Request("http://127.0.0.1:5000/potentialMatch", {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify({ responses: this.state.response, userId: id }),
+          });
+
+        fetch(myRequest)
+          .then((res) => res.json())
+          .then((res) => {
+            if (res.response === "Success") {
+                //Once the first questionnaire API call is a success, the second matching call is nested to ensure that the two calls
+                //happen in succession
+                fetch(matchRequest)
+                  .then((res) => res.json())
+                  .then((res) => {
+                    if (res.response === "Success") {
+                        this.setState({
+                          error: "Successfully updated your questionnaire responses!",
+                        });
+                    } else {
+                      this.setState({
+                        error: res.response,
+                      });
+                    }
+                  });
+                
+          } else {
+            this.setState({
+              error: res.response,
+            });
+          }
+        })
+
+        .catch((error) => {
+          this.setState({
+            error: "Error connecting to backend",
+          });
+        });
+    } else {
+      alert("Please answer all questions");
+    }
+  }
+  checkResponses(){
+    var countanswer=0;
+    this.state.response.map(answer=>{
+      if(answer == 0 || answer == 1){
+        countanswer++
+      }
+    })
+    return countanswer==16
+  }
+  getQuestions = () => {
+    questions().then((question) => {
+      this.setState({
+        questionBank: question,
+      });
+    });
+  };
+  componentDidMount() {
+    this.getQuestions();
+  }
+  storeAnswer = (answer, ID) => {
+    this.state.response[ID - 1] = answer;
+  };
+  render() {
+    if (this.state.id === "") {
+      this.props.history.push({
+        pathname: "/login",
+      });
+      return null;
+    }
+    return (
+      <div>
+          <Header id={this.state.userId} />
```

Comment:
If you're using the same HTML/jsx for the `updateQuestionnaire.js` as the `Questionnaire.js`  then you can add the jsx into one file that can be rendered within both files to reduce redundant code

Reply:
Actually not sure if that's possible ^ with our current state management lol